plutus-scb/src/Plutus/SCB/Types.hs

```diff
@@ -101,4 +105,28 @@ data ChainOverview =
     deriving (Show, Eq, Generic)
     deriving anyclass (ToJSON, FromJSON)
 
+mkChainOverview :: Blockchain -> ChainOverview
+mkChainOverview = foldl reducer emptyChainOverview
+  where
+    reducer :: ChainOverview -> Block -> ChainOverview
+    reducer ChainOverview { chainOverviewBlockchain = oldBlockchain
+                          , chainOverviewUnspentTxsById = oldTxById
+                          , chainOverviewUtxoIndex = oldUtxoIndex
+                          } txs =
+        let unprunedTxById =
+                foldl (\m tx -> Map.insert (txId tx) tx m) oldTxById txs
+            newTxById = id unprunedTxById -- TODO Prune spent keys.
+            newUtxoIndex = UtxoIndex.insertBlock txs oldUtxoIndex
+         in ChainOverview
+                { chainOverviewBlockchain = txs : oldBlockchain
+                , chainOverviewUnspentTxsById = newTxById
+                , chainOverviewUtxoIndex = newUtxoIndex
+                }
+    emptyChainOverview =
+        ChainOverview
+            { chainOverviewBlockchain = []
+            , chainOverviewUnspentTxsById = Map.empty
+            , chainOverviewUtxoIndex = UtxoIndex Map.empty
+            }
```

Comment:
I'm not sure they can be derived. (If it were a PureScript record, I'd say yes, but this is Haskell.)

Even so...this is a case where we could make a _legal_ Monoid instance, but I'm not convinced we could make a _meaningful_ one. We'd essentially end up finding out which `chainOverviewBlockchain` was newer, and then throwing the other overview away. 

(That might suggest it's worth putting an `Ord` instance on them, maybe. But I can't foresee a case where we have two instances of this at the same time to compare or combine.)

Reply:
There's `GenericMonoid`/`GenericSemigroup` from `semigroups` which you can derive via if you like.

Reply:
That's interesting, I saw this as appending over time or accumulating an overview over time. Basically since you are unioning the maps and consing the list, I don't see how that isn't meaningful. The resulting overview is the total of those that came before it. Then again I suppose it's like a materialized view of the blockchain, so it's the blockchain that's being appended, not the overview.